lib/logstash/outputs/elasticsearch.rb

```diff
@@ -368,9 +370,31 @@ def submit(actions)
   def flush(actions, teardown = false)
     begin
       submit(actions)
+    rescue Manticore::SocketException => e
+      # If we can't even connect to the server let's just print out the URL (:host is actually a URL)
+      # and let the user sort it out from there
+      @logger.error(
+        "Attempted to send a bulk request to Elasticsearch configured at '#{@current_client.client_options[:host]}',"+
+          " but Elasticsearch appears to be unreachable or down!",
+        :client_config => @current_client.client_options,
+        :error_message => e.message,
+        :error_class => e.class.name,
+        :backtrace => e.backtrace
```

Comment:
My rationale here is that this is a very common error that shows up in your support tickets often. I think its hard to evaluate it if there's any ambiguity in the messages, so including more info helps us troubleshoot issues for our users more quickly.

Additionally, people tend to blame logstash, but having the backtrace makes it more apparent more is at work here.